v2/export.go

```diff
@@ -0,0 +1,101 @@
+package iavl
+
+import "fmt"
+
+// TraverseOrderType is the type of the order in which the tree is traversed.
+type TraverseOrderType uint8
+
+const (
+	PreOrder TraverseOrderType = iota
+	PostOrder
+)
+
+type Exporter struct {
+	tree  *Tree
+	out   chan *Node
+	errCh chan error
+}
+
+func (tree *Tree) Export(order TraverseOrderType) *Exporter {
+	exporter := &Exporter{
+		tree:  tree,
+		out:   make(chan *Node),
```

Comment:
To speed this code up, please make these buffered channels with some capacity by means of a heuristic, so that traversal won't be constrained and blocked by the speed of reading out the nodes